presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1183,6 +1185,21 @@ private PhysicalOperation visitScanFilterAndProject(
                 sourceLayout = source.getLayout();
             }
 
+            // filterExpression may contain large function calls; evaluate them before compiling.
+            if (filterExpression.isPresent()) {
+                Type type = filterExpression.get().getType();
```

Comment:
Filter can only be Boolean?

Reply:
I would keep the original type to avoid any logic injection.

Reply:
Yes, this makes sense then.